gnd/src/androidTest/java/com/google/android/gnd/persistence/sync/FakeWorkManager.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.persistence.sync;
+
+import android.app.PendingIntent;
+import androidx.annotation.NonNull;
+import androidx.lifecycle.LiveData;
+import androidx.work.ExistingPeriodicWorkPolicy;
+import androidx.work.ExistingWorkPolicy;
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.Operation;
+import androidx.work.PeriodicWorkRequest;
+import androidx.work.WorkContinuation;
+import androidx.work.WorkInfo;
+import androidx.work.WorkManager;
+import androidx.work.WorkQuery;
+import androidx.work.WorkRequest;
+import com.google.common.util.concurrent.ListenableFuture;
+import java.util.List;
+import java.util.UUID;
+
+public class FakeWorkManager extends WorkManager {
```

Comment:
Could you kindly remind me why Mockito mocks don't work here? It seems this fake has the same behavior as the Mockito's mock. Any reason we can't just use that?

Reply:
I tried using mockito first. But the mocks couldn't be generated. It expects a proper test runner.

Reply:
Sgtm. I don't really understand the relationship between test runner and providing a worker - let's discuss offline?